common/fixation_migrate.go

```diff
@@ -38,6 +41,39 @@ func (fs *FixationStore) MigrateVersion(ctx sdk.Context) (err error) {
 	return nil
 }
 
+// MigratePrefix replaces objects' prefixes from `oldPrefix` to the current `fs.prefix`
+// NOTE: it first tries to perform fs.MigrateVersion() with the old prefix.
+func (fs *FixationStore) MigratePrefix(ctx sdk.Context, oldPrefix string) (err error) {
+	newPrefix := fs.prefix
+
+	// first check if version upgrade is due - must use old prefix
+	fs.prefix = oldPrefix
+	err = fs.MigrateVersion(ctx)
+	fs.prefix = newPrefix
+
+	if err != nil {
+		return err
+	}
+
+	// prefix migration
+	fs.doMigratePrefix(ctx, oldPrefix, newPrefix)
```

Comment:
why do we need to migrate a second time?

Reply:
Usure what you mean?

`fs.MigratePrefix()` is an API for users of FixationStore that wants to change their prefix, i.e. a data migration requested by the module that uses it.  It calls `fs.doMigratePrefix()` to do the actual work. But first it calls `fs.MigrateVersion()` to apply any internal updates of `fs.MigratePrefix()` (i.e. a migration decided/needed by the FixationStore) - and it does so using the *old* prefix (otherwise the *new* prefix would be used before it's in effect).